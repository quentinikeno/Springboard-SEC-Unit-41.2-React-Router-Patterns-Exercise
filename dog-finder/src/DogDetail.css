.DogDetail {
	width: fit-content;
	padding: 1rem 4rem;
	background-color: #f5f5f5;
	border-radius: 20px;
	box-shadow: 10px 10px 10px rgba(0, 0, 0, 0.25);
	margin: 1rem auto;
}
.DogDetail img {
	max-width: 1000px;
	max-height: 1000px;
	border-radius: 50%;
}

.DogDetail ul {
	width: fit-content;
	margin: 0 auto;
	text-align: left;
}

.DogDetail-facts {
	margin-bottom: 0;
}
