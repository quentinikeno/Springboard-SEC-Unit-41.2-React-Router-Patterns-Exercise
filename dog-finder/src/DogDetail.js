import { useParams } from "react-router-dom";
import "./DogDetail.css";

const DogDetail = ({ dogs }) => {
	const { name } = useParams();
	const foundDog = dogs.find(
		(dog) => dog.name.toLowerCase() === name.toLowerCase()
	);
	const { name: dogName, age, src, facts } = foundDog;
	return (
		<div className="DogDetail">
			<h1>{dogName}</h1>
			<img src={src} alt={name} />
			<h4>Age: {age}</h4>
			<h4 className="DogDetail-facts">Facts:</h4>
			<ul>
				{facts.map((fact) => (
					<li key={fact}>{fact}</li>
				))}
			</ul>
		</div>
	);
};

export default DogDetail;
