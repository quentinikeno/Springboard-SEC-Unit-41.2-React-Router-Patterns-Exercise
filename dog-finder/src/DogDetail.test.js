import { render } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router-dom";
import DogDetail from "./DogDetail";
import dogs from "./dogs";

it("should render without crashing", () => {
	render(
		<MemoryRouter>
			<Route path="/dogs/:name">
				<DogDetail dogs={dogs} />
			</Route>
		</MemoryRouter>,
		{
			route: "/dogs/tubby",
		}
	);
});

it("should matchSnapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<Route path="/dogs/:name">
				<DogDetail dogs={dogs} />
			</Route>
		</MemoryRouter>,
		{
			route: "/dogs/tubby",
		}
	);
	expect(asFragment()).toMatchSnapshot();
});
