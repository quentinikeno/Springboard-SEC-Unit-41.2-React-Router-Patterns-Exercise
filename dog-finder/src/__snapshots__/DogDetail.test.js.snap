// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matchSnapshot 1`] = `<DocumentFragment />`;
